src/components/[guild]/Requirements/RoleRequirements.tsx

```diff
@@ -40,12 +41,13 @@ const RoleRequirements = ({
   initialRequirementsRef,
 }: Props) => {
   const guild = useGuild()
+  const { data } = useRequirements(role.id)
 
   const requirements =
     role.hiddenRequirements ||
-    ((role.requirements ?? []).length === 0 && !(guild as any).isFallback)
-      ? [...role.requirements, { type: "HIDDEN", roleId: role.id } as Requirement]
-      : role.requirements
+    ((data ?? []).length === 0 && !(guild as any).isFallback)
```

Comment:
Nit: Noticing a lot of `data?... ?? []` and `data ?? []` patterns, maybe it is worth considering for `useRequirements` to always return an array

Reply:
Hmm, not sure about this, I think in general data should be `undefined` until we don't fetch anything (or if we encounter an error), so I wouldn't like to assign a default value to it, but I assigned a default value to `data` in this component only, so the code is a bit cleaner now ;)